add doc for v5.1: Deploy -> Operate TiKV (#232)
tikv/website: content/docs/5.1/deploy/operate/scale.md

---
title: Scale
description: Learn how to scale out/in TiKV using TiUP
menu:
    "5.1":
        parent: Operate TiKV
        weight: 2
---

The capacity of a TiKV cluster can be increased or decreased without interrupting the online services.

This document describes how to scale the TiKV and PD using TiUP.

> 

Reviewer: ```suggestion

## View a TiKV cluster

```

Author: `View a TiKV cluster` is not a chapter of this page.
It is mentioned in case the user doesn't know how the cluster name.

> To view the current cluster name list, run `tiup cluster list`.

Reviewer: what is the name list?

Author: it's not necessary to list the cluster names here.

Reviewer: I mean, what does `name list` means?

Reviewer 2: we can assume that there is only one target cluster to be scaled.

Author: this line is a little bit strange.

> For example, if the original topology of the cluster is as follows:

Reviewer: ```suggestion
For example, assume that the topology of the cluster is as follows:
```

| Host IP  | Service |
|:-------- |:------- |
| 10.0.1.1 | Monitor |
| 10.0.1.2 | PD      |
| 10.0.1.3 | PD      |
| 10.0.1.4 | PD      |
| 10.0.1.5 | TiKV    |
| 10.0.1.6 | TiKV    |
| 10.0.1.7 | TiKV    |

## Scale out a TiKV cluster

If you want to add a TiKV node to the `10.0.1.8` host, take the following steps.

{{< info >}}
You can take similar steps to add the PD node.
{{< /info >}}

1. Configure the scale-out topology:

>     Add the scale-out topology configuration in the `scale-out.yaml` file:
> 
>     ```bash
>     vi scale-out.yaml
>     ```

Reviewer 2: ```suggestion
    Put the following content in the `scale-out.yaml` file:
```

    Here is a TiKV configuration file template:

    ```yaml
    tikv_servers:
    - host: 10.0.1.8
        ssh_port: 22
        port: 20160
        status_port: 20180
        deploy_dir: /data/deploy/install/deploy/tikv-20160
        data_dir: /data/deploy/install/data/tikv-20160
        log_dir: /data/deploy/install/log/tikv-20160
    ```

>     Here is a PD configuration file template:

Reviewer 2: better to add another section to describe how to scale out pd.

Author: good idea

Author: done

    ```yaml
    pd_servers:
    - host: 10.0.1.9
        ssh_port: 22
        client_port: 2379
        peer_port: 2380
        deploy_dir: /data/deploy/install/deploy/pd-2379
        data_dir: /data/deploy/install/data/pd-2379
        log_dir: /data/deploy/install/log/pd-2379
    ```

    To view the configuration of the current cluster, run `tiup cluster edit-config <cluster-name>`. Because the parameter configuration of `global` and `server_configs` is inherited by `scale-out.yaml` and thus also takes effect in `scale-out.yaml`.

2. Run the scale-out command:

    ```shell

>     tiup cluster scale-out <cluster-name> scale-out.yaml

Reviewer 2: after scaling out, does the user need to change their cluster topology as well?

Author: no, users do not need to

Author: `topology.yaml`  `scale-out.yaml` are only used by the `tiup cluster` command and they can be deleted after that.
